### ISSUE TITLE
Missing zip codes?

### ISSUE THREAD

**Issue by hayksaakian** [blakek/us-zips#43]
I had some code bug out because certain zip codes were missing when I used this library.

Example: 49588

**Comment by blakek** [comment]
**TL;DR:** This is a known issue, but I don't know where to get more accurate information.

I acknowledge the missing ZIP codes and understand your concern. This package uses Census Bureau [ZIP Code Tabulation Areas](https://www.census.gov/programs-surveys/geography/guidance/geo-areas/zctas.html), which do not include every USPS ZIP code. Unfortunately, I'm unaware of an official, up-to-date USPS ZIP code list due to their frequent changes for mail routes. There is a [note in the README](https://github.com/blakek/us-zips/blob/d2665c813cc0ae626feacfab455ffeff9747ecd5/README.md?plain=1#L16-L17) that some ZIP codes won't be available.

For the specific example you provided (49588), Google Maps does show an area southeast of Grand Rapids, MI, which seems like it'd contain several addresses:

<img width="260" alt="" src="https://user-images.githubusercontent.com/4932702/230981191-47123eea-270e-4b0c-a2ea-456ae42b0905.png">

However, looking at addresses within that boundary largely shows other ZIPs like 49508 or 49512 (which are both listed in this package):

<img width="567" alt="" src="https://user-images.githubusercontent.com/4932702/230981351-a6e4c6cc-884f-4259-88d4-5c53610f84d9.png">

To be clear, I agree that there are ZIP codes used by USPS that will not show here. To my knowledge, there is no official list of up-to-date ZIP codes used by USPS. The ZIP lists are managed locally and change often; they are intended to be used for mail routes only. This package makes tradeoffs of being:

- fast for lookups but not exact (uses center of ZCTAs)
- containing _most_ residential addresses
- up-to-date (an issue is automatically created when the Census Bureau publishes updated ZCTA data; old ZIPs are removed)

**Comment by hayksaakian** [comment]
I appreciate the effort and clarification here. 

I later found a dataset here that at least identifies the zip codes, and has some record of when it was updated

https://postalpro.usps.com/ZIP_Locale_Detail

It doesn't really have any latitude or longitude information, so it's a starting point more than anything.
